src/hooks/useAgreementForm.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+
+import { GRAASP_LANDING_PAGE_HOST } from '../config/env';
+import { useAuthTranslation } from '../config/i18n';
+import { AUTH } from '../langs/constants';
+
+export type UseAgreementForm = {
+  userHasAcceptAllTerms: boolean;
+  updateUserAgreements: (hasAgree: boolean) => void;
+  verifyUserAgreements: () => boolean;
+  hasError: boolean;
+  termsOfServiceLink: string;
+  privacyPolicyLink: string;
+};
+
+export const useAgreementForm = (): UseAgreementForm => {
+  const { t, i18n } = useAuthTranslation();
+
+  const [userHasAcceptAllTerms, setUserHasAcceptAllTerms] = useState(false);
+  const [termsOfServiceLink, setTermsOfServiceLink] = useState(
+    t(AUTH.TERMS_OF_SERVICE_LINK),
+  );
+  const [privacyPolicyLink, setPrivacyPolicyLink] = useState(
+    t(AUTH.PRIVACY_POLICY_LINK),
+  );
```

Comment:
Not sure these are required to be states. I see you have to update them with useEffects, maybe they can be simple variables ?

Reply:
I tried with useRef and I have a sync problem. For example, if I change the language like the following:

0. English (default set lang)
1. French
2. German
3. French
4. Italian

I got:

0. en
1. fr
2. fr
3. en (because german doesn't have the links for now)
4. fr

When using a simple variable, it seems to work, but I have the following warning:
`Assignments to the 'privacyPolicyLink' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.`.

Reply:
Ok, I updated as you suggested Basile and it works perfectly, thanks !